core/src/main/kotlin/org/evomaster/core/output/dto/DtoWriter.kt

```diff
@@ -56,29 +56,66 @@ class DtoWriter {
         actionDefinitions.forEach { action ->
             action.getViewOfChildren().find { it is BodyParam }
             .let {
-                val primaryGene = (it as BodyParam).primaryGene().getLeafGene()
-                // TODO: Payloads could also be json arrays, analyze ArrayGene
-                when (primaryGene) {
-                    is ObjectGene -> {
-                        calculateDtoFromObject(primaryGene, action.getName())
-                    }
-                    is ArrayGene<*> -> {
-                        calculateDtoFromArray(primaryGene, action.getName())
-                    }
-                    else -> {
-                        throw IllegalStateException("Gene $primaryGene is not supported for DTO payloads")
+                val primaryGene = (it as BodyParam).primaryGene()
```

Comment:
what happens if the body is optional, and the primary gene is a wrapper `OptionalGene`?

Reply:
Good question, which would be the best way to handle those cases? I saw you removed the `getWrappedValueGene` function. The `getLeafNode` of a `ChoiceGene` wrapped in an `OptionalGene` returns only one of the options

Reply:
@arcuri82

Reply:
if it is a `ChoiceGene` inside an `OptionalGene`, you will not see it here. `ChoiceGene` complicates a few things, not just here. so in long run i need to think about some new functionality/fixes for it. for now, maybe an approach could be that, on `primaryGene`, you do a call on `getWrappedGene(ChoiceGene::class.java)`, and see if null or not